### PR TITLE
fix: feature propagation

### PR DESCRIPTION
**Pull request by liamaharon**
Feature propagation is broken for a bunch of crates, breaking compiling in some scenarios. e.g. `cargo c -p reth-auto-seal-consensus --features optimism` fails due to the crate incorrectly propagating the `optimism` feature.

This PR fixes feature propagation across crates and adds a new lint CI step to keep it consistent.

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
cool, all of these look good and having zepter makes it easy to not violate this again

just need to check how long this takes in ci

**Review by DaniPopes — APPROVED** [review]
Looks very useful, thanks!
Compiling takes 1 minute, and running takes 1 second. This is fine.
For reference: [`zepter`](https://github.com/ggwpez/zepter)

**Comment by liamaharon** [comment]
Fixed conflicts and added `serde,test-utils,arbitrary,bench` to checked features

**Comment by liamaharon** [comment]
Fixed conflicts again :)

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -219,6 +219,22 @@ jobs:
         env:
           RUSTFLAGS: -D warnings
 
+  # Check crates correclty propagate features
```
Comment (onbjerg) [inline .github/workflows/lint.yml:222]:
```suggestion
  # Check crates correctly propagate features
```

**Pull request merged**